.github/workflows/ci-linux.yml

```diff
@@ -59,7 +76,7 @@ jobs:
         uses: actions/cache/restore@v3
         with:
           path: ${{ env.CACHE_DIR }}
-          key: linux-build-test-cpp-asserts-manylinux-v2-${{ github.sha }}
+          key:  ${{ env.CACHE_KEY }}
           restore-keys: linux-build-test-cpp-
```

Comment:
> FYI for anyone wondering how this all works - the [restore-keys](https://github.com/nod-ai/iree-amd-aie/blob/main/.github/workflows/ci-linux.yml#L63) are longest prefix matched; so

I might be missing something, but [here](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#matching-a-cache-key) it says:

> When a key doesn't match directly, the action searches for keys prefixed with the restore key. If there are multiple partial matches for a restore key, the action returns the most recently created cache.

Still, it would first look for an exact match with `key` and only search further if not found, so I think this behaves like you're intending.

Reply:
Ctrl+f prefix here https://github.com/actions/cache?tab=readme-ov-file#inputs